src/App.js

```diff
@@ -14,11 +16,124 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state = {
+    goods: goodsFromServer,
+    isHidden: true,
+    isReversed: false,
+    sortBy: '',
+  }
+
+  buttonSatus = () => {
+    this.setState({
+      isHidden: false,
+    });
+  }
+
+  reverseMethod = () => {
+    this.setState(state => ({
+      isReversed: !state.isReversed,
```

Comment:
It would be great to move all the sortings inside the `GoodsList` component.
It's good to split the logic. `App` component should only have the button 'start' which will toggle the render of the `GoodsList`. There you should pass initial goods and sort them inside that component (not inside the `App`)